Content.Server/Audio/Jukebox/JukeboxSystem.cs

```diff
@@ -153,4 +224,29 @@ private void TryUpdateVisualState(EntityUid uid, JukeboxComponent? jukeboxCompon
 
         _appearanceSystem.SetData(uid, JukeboxVisuals.VisualState, finalState);
     }
+
+    private JukeboxBoundUserInterfaceState GetUiState(Entity<JukeboxComponent> ent)
+    {
+        var isPlaying = false;
+        if (Exists(ent.Comp.AudioStream))
+        {
+            isPlaying = Audio.IsPlaying(ent.Comp.AudioStream);
+        }
+        return new JukeboxBoundUserInterfaceState(ent.Comp.SongIdQueue, isPlaying);
+    }
+
+    private ProtoId<JukeboxPrototype>? SongQueueDequeue(Entity<JukeboxComponent> ent)
+    {
+        if (ent.Comp.SongIdQueue.Count == 0)
+            return null;
+
+        var next = ent.Comp.SongIdQueue[0];
+        ent.Comp.SongIdQueue.RemoveAt(0);
+        ent.Comp.TimeWhenSongEnds = null;
+
```

Comment:
This is going to shuffle the entire list, just use an appropriate collection type instead like an actual queue.

Reply:
For now I changed `SongIdQueue` to Queue as there is no LinkedList serializer in the engine.